### PR TITLE
[ARTEMIS-4238]: transaction-timeout ActivationConfigProperty is no lo…

### PR DESCRIPTION
**Pull request by ehsavoie**
…nger working.

* Re-adding the missing code

Jira: https://issues.apache.org/jira/browse/ARTEMIS-4238

### CODE REVIEW

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/ActiveMQRAManagedConnection.java [hunk 1]
```diff
@@ -336,7 +336,7 @@ public void associateConnection(final Object obj) throws ResourceException {
 
    public void checkTransactionActive() throws JMSException {
       // don't bother looking at the transaction if there's an active XID
-      if (!inManagedTx && tsr != null) {
+      if (!inManagedTx && tsr != null && tsr.getTransactionKey() != null) {
```
Comment (jbertram) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/ActiveMQRAManagedConnection.java:339]: This change doesn't appear related to the `transactionTimeout` change. Why is this necessary?

**Comment by jbertram** [comment]
It would be ideal to have some tests for this to catch future regressions.

**Comment by jbertram** [comment]
The commit message doesn't follow the [recommended format](https://github.com/apache/activemq-artemis/blob/main/docs/hacking-guide/en/code.md#commitMessageDetails). There's no need for square brackets and a colon around the Jira reference.

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/ActiveMQRAManagedConnection.java (hunk 1 above)
Comment (ehsavoie) [reply artemis-ra/src/main/java/org/apache/activemq/artemis/ra/ActiveMQRAManagedConnection.java:339]: a leftover

**Comment by jbertram** [comment]
@ehsavoie, this looks good. I transitioned it from draft and will merge shortly.
